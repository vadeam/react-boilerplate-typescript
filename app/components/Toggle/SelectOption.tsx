/**
 * SelectOption
 */

import * as React from 'react'
import { injectIntl, MessageDescriptor, WrappedComponentProps } from 'react-intl'

type SelectOptionProps = { key: string; value: string | number; message?: MessageDescriptor } & WrappedComponentProps

const SelectOption = ({ value, key, message, intl }: SelectOptionProps) => (
  <option value={value} key={key}>
    {message ? intl.formatMessage(message) : value}
  </option>
)

export default injectIntl(SelectOption)
