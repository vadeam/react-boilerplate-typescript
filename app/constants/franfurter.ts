export const DEFAULT_BASE_CURRENCY = 'EUR'
