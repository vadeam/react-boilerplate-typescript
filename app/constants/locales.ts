export const LOCALE_DE = 'de'
export const LOCALE_EN = 'en'
export const DEFAULT_LOCALE = LOCALE_EN

export const appLocales: Locales[] = [LOCALE_EN, LOCALE_DE]
export const CUSTOM_LOCALE_STORAGE_KEY = 'custom-locale'
