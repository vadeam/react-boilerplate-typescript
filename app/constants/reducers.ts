export const REDUCER_KEY_APP = 'app'
export const REDUCER_KEY_LANGUAGE = 'language'
export const REDUCER_KEY_ROUTER = 'router'
export const REDUCER_KEY_HOME = 'home'
// for testing purposes
export const REDUCER_KEY_TEST = 'test'
