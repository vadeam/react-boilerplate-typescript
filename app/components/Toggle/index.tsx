/**
 * Toggle
 */

import * as React from 'react'
import { MessageDescriptor } from 'react-intl'

import SelectOption from './SelectOption'
import Select from './Select'

interface ToggleProps {
  onToggle: (event: any) => void
  messages?: {
    [k: string]: MessageDescriptor
  }
  value?: string | number
  values?: string[]
}

function Toggle({ values, value, messages, onToggle }: ToggleProps) {
  const emptyValue = '--'
  let content: JSX.Element | JSX.Element[] = <SelectOption key={emptyValue} value={emptyValue} />

  // If we have items, render them
  if (values) {
    content = values.map((val) => <SelectOption key={val} value={val} message={messages ? messages[val] : undefined} />)
  }

  return (
    <Select value={value ?? emptyValue} onChange={onToggle}>
      {content}
    </Select>
  )
}

export default Toggle
