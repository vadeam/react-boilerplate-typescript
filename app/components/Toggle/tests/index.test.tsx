import { render } from '@testing-library/react'
import React from 'react'
import { IntlProvider, defineMessages } from 'react-intl'

import Toggle from '../index'
import { appLocales, DEFAULT_LOCALE } from '../../../constants/locales'

describe('<Toggle />', () => {
  it('should contain default text', () => {
    const messages = defineMessages(
      appLocales.reduce(
        (locales, locale) => ({
          ...locales,
          [locale]: {
            id: `containers.LocaleToggle.${locale}`,
            defaultMessage: `default ${locale.toUpperCase()} message`,
          },
        }),
        {},
      ),
    )
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Toggle values={appLocales} messages={messages} onToggle={() => null} />
      </IntlProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should not have ToggleOptions if props.values is not defined', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Toggle onToggle={() => null} />
      </IntlProvider>,
    )
    const elements = container.querySelectorAll('option')
    expect(elements).toHaveLength(1)
    expect(container.firstChild).toHaveTextContent('--')
  })
})
