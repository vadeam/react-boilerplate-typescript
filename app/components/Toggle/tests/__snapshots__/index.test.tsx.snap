// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should contain default text 1`] = `
<select
  class="sc-bdfBQB igqybA"
>
  <option
    value="en"
  >
    default EN message
  </option>
  <option
    value="de"
  >
    default DE message
  </option>
</select>
`;
